src/ansible_builder/user_definition.py

```diff
@@ -243,3 +245,13 @@ def validate(self):
                     self.build_arg_defaults['EE_BUILDER_IMAGE'] = self.builder_image.name
 
             self._validate_additional_build_files()
+
+            if self.version == 3:
+                build_steps = self.raw.get('additional_build_steps', {})
+                for step_name, steps in build_steps.items():
+                    for directive in steps:
+                        if directive.startswith('USER '):
+                            logging.warning(
+                                f"Found USER directive in '{step_name}' in 'additional_build_steps'. "
+                                f"Remove this directive from '{step_name}', "
+                                f"since this may cause failure in the build process.")
```

Comment:
The `Remove this directive` part sounds more like a demand rather than an informative suggestion which would seem more appropriate for a warning message. Maybe something more like "Including this directive may cause failures in the build process."

Also, can we add a test that captures the output (maybe with `capsys`?) and verifies this gets output to the user?